MapboxCoreNavigation/Constants.swift

```diff
@@ -92,3 +92,8 @@ let FasterRouteFoundEvent = "navigation.fasterRoute"
  The number of seconds remaining on the final step of a leg before the user is considered "arrived".
  */
 public var RouteControllerDurationRemainingWaypointArrival: TimeInterval = 3
+
+/**
+ The minimum distance of a tunnel intersection on a given route step.
+ */
+public var RouteProgressMinimumTunnelIntersectionDistance: CLLocationDistance = 100
```

Comment:
Do you think 100&nbsp;meters is an appropriate distance? I don’t have a good intuition about metric distances, but on a surface street, a tunnel much shorter than [this](https://www.youtube.com/watch?v=ew74X5z3Rgo) would still seem like plenty for switching to the night style. Per https://github.com/mapbox/mapbox-navigation-ios/pull/1127#issuecomment-370557811, factoring in the current speed would help us avoid special cases for freeways versus surface streets. SimulatedLocationManager has some code to determine the expected speed of a given segment along a step:

https://github.com/mapbox/mapbox-navigation-ios/blob/b381563387262b408dfe1939883c813bd616654b/MapboxCoreNavigation/SimulatedLocationManager.swift#L130-L134